app/presenters/provider_oauth_flow_presenter.rb

```diff
@@ -1,4 +1,4 @@
-class ProviderOauthFlowPresenter < OauthFlowPresenter
+class ProviderOAuthFlowPresenter < OAuthFlowPresenter
```

Comment:
Is there a special rule about `OAuth`? Because usually only first letter would be capitalized.

Reply:
We don't need it.

For Rails 5.2, the autoloader will find `ProviderOAuthFlowPresenter` and will look for a file called `"ProviderOAuthFlowPresenter".underscore`, that is, `provider_oauth_flow_presenter`. Which matches the name of the file.

For Rails 6, it will find a file called `provider_oauth_flow_presenter` and will expect `"provider_oauth_flow_presenter".camelize` => `ProviderOAuthFlowPresenter` to be defined inside. I know it's strange but check this in the rails console:

```
[4] pry(main)> "oauth".camelize
=> "OAuth"
[5] pry(main)> "oauth2".camelize
=> "Oauth2"
```

The problem only exists for "oauth2", for which we have defined some [inflectors](https://github.com/3scale/porta/blob/THREESCALE-7405-update-rails-to-6-0/config/initializers/zeitwerk.rb) in rails 6.

`camelize` makes other exceptions, like:

```
[16] pry(main)> "github".camelize
=> "GitHub"
[17] pry(main)> "sso".camelize
=> "SSO"
```

Reply:
Yeah, this is what I meant. I didn't expect `"oauth".camelize` -> `"OAuth"`

It's nice.

Reply:
This is because of https://github.com/3scale/porta/blob/d11cf2129b7fe952b5073dd364e3ea3f187543ac/config/initializers/inflections.rb#L22

By default, Rails would camelize "oauth" as "Oauth", but porta has a special rule for OAuth (as well as some other acronyms)

Reply:
Thanks for the info!